src/dynamic_boundary_conditions/rainfall_sites.py

```diff
@@ -46,7 +46,14 @@ def get_rainfall_sites_data() -> gpd.GeoDataFrame:
     headers["User-Agent"] = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko)\
         Chrome/96.0.4664.45 Safari/537.36"
     response = requests.get(url, headers=headers)
-    sites_df = pd.read_json(response.text)
+    sites_data = response.text
+    return sites_data
+
+
+def get_rainfall_sites_in_df() -> gpd.GeoDataFrame:
+    """Get rainfall sites data from the HIRDS website and transform to GeoDataFrame format."""
+    sites_data = get_rainfall_sites_data()
+    sites_df = pd.read_json(sites_data)
```

Comment:
I initially defined this function as you suggested, but I thought it was better to do it the other way so that only this function needs to get called to fetch the data.